src/context.rs

```diff
@@ -108,12 +111,15 @@ mod alloc_only {
     const ALIGN_TO: usize = ::core::mem::align_of::<AlignedType>();
 
     /// Represents the set of capabilities needed for signing.
+    #[cfg_attr(docsrs, doc(cfg(any(feature = "std", feature = "alloc"))))]
```

Comment:
(philosophical code cleanliness discussion :)

You put the attribute _below_ the rustdoc here. 

A few points in no particular order:

- As your eyes scan down a file the most important thing should come first, the rustdoc comment is more important than the attribute. Supports comment above attribute.

- When one reads a function your eye flicks back an forth between the code and comment _if_ one is reading the comment. The eye movement is easier if there is not the clutter of the attribute in between. Supports comment below the attribute.

- Rust idiomatic style is???

- Seems typical in this project to put attribute above the rustdoc but I could be wrong.

Reply:
I personally prefer to put attributes under doc comments but I'm not sure why. Intuition, I guess?

Reply:
A quick bit of research tells me that the idiomatic ordering is inline with both of the comments above, comments first then attributes

ref: 
- https://doc.rust-lang.org/reference/attributes.html
- A bunch of 'random' files from the Rust source code

Reply:
Another review of this PR showed me you were correct and uniform @Kixunil, I was wondering _why_ I brought this up. I did a [PR](https://github.com/rust-bitcoin/rust-bitcoin/pull/753) to try and work it out but could not, sorry to derail your PR :)